### DISCUSSION TITLE
Received an error while running bpftrace

### DISCUSSION THREAD

**Discussion by Engg-blogger** [bpftrace/bpftrace#3247] [General]
Hi Team - I was running bpftrace on one of the rpm based distro and faced an error "'/tmp/kheaders-5.15.153.1-2.cm2/include/linux/kconfig.h' file not found". 
It was kind of weird because bpftrace was already running on this vm, but after the service got restarted, this error started coming. After restart of the vm, error went away. Could you please suggest the reason behind this intermittent issue?

**Comment by viktormalik** [comment]
Hi, could you please give us some details?

- What was the version of bpftrace that you were trying to run?
- Could you share the script that you were trying to run?
- What is the mentioned service?

Thanks!

**Reply by Engg-blogger** [reply]
The version of bpftrace is v0.20.0. Script is tcpconnect.bt. I am wondering if bpftrace appimage creates any files in tmp folder that needs to be cleaned up after closing the script?

**Reply by viktormalik** [reply]
If you don't have the kernel development package installed, bpftrace 0.20.0 tries to unpack the kernel headers from `/sys/kernel/kheaders.tar.xz` into `/tmp/kheaders-$(uname -r)`. It is possible that you upgraded the kernel and therefore the version didn't match.

Anyways, the header unpacking has been removed in bpftrace 0.20.4 as it contained a vulnerability. I strongly recommend upgrading to that version. Also, the latest version of tcpconnect.bt (from master) doesn't need kernel headers at all and relies on BTF only.

**Reply by Engg-blogger** [reply]
thanks :)
